### PR TITLE
Extra headers for bedrock

### PR DESCRIPTION
**Pull request by themrzmaster**
When using a proxy in front of bedrock, it might be necessary to pass extra headers to the endpoint url.
This PRs allows it.
Also, it allows some other data types to be on the message list

### CODE REVIEW

File: litellm/llms/bedrock.py [hunk 1]
```diff
@@ -504,12 +513,12 @@ def init_bedrock_client(
     aws_session_name: Optional[str] = None,
     aws_profile_name: Optional[str] = None,
     aws_role_name: Optional[str] = None,
+    extra_headers: Optional[dict] = None,
```
Comment (ghost) [inline litellm/llms/bedrock.py]: @themrzmaster we already have support for something similar via metadata, see how langfuse logging works - https://docs.litellm.ai/docs/observability/langfuse_integration#set-custom-trace-id-trace-user-id-and-tags

couldn't we just use the metadata field here?

Comment (ghost) [reply litellm/llms/bedrock.py]: sounds like the purpose of extra headers is for bedrock logging?

Comment (themrzmaster) [reply litellm/llms/bedrock.py]: in my scenario, I use bedrock over a custom proxy. i need to pass custom authorization header to be able to access the endpoint

File: litellm/utils.py [hunk 2]
```diff
@@ -5143,7 +5146,7 @@ def get_supported_openai_params(model: str, custom_llm_provider: str):
     """
     if custom_llm_provider == "bedrock":
         if model.startswith("anthropic.claude-3"):
-            return litellm.AmazonAnthropicClaude3Config().get_supported_openai_params()
+            return litellm.AmazonAnthropicClaude3Config().get_supported_openai_params() + ["extra_headers"]
```
Comment (ghost) [inline litellm/utils.py]: @themrzmaster can you remove this line. 

It'll send the extra headers as part of the request body during calls

Comment (ghost) [reply litellm/utils.py]: you've already solved your problem by taking it in as a direct param into the completion() function in main.py

Comment (themrzmaster) [reply litellm/utils.py]: if i remove, i get 
` bedrock does not support parameters: {'extra_headers'}`
@krrishdholakia

Comment (themrzmaster) [reply litellm/utils.py]: any other idea how to avoid this error?

Comment (themrzmaster) [reply litellm/utils.py]: @krrishdholakia it would really help me if we could ship that soon. Is there anything I can do?

Comment (themrzmaster) [reply litellm/utils.py]: i changed it to bedrock.py

**Pull request closed without merging**